### PR TITLE
#4: Rust API unit tests

### PR DESCRIPTION
**Pull request by joao-conde**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/pconvert-rust/issues/4 |
| Dependencies | None |
| Testing | `cargo test` at the root level of the repo |
| Decisions | In this PR, we make the command-line "tests" we had also runnable via `cargo test`. The code to execute is the same, hence we refactor common code into tests and use it in the command-line binary. Note that the reverse can't be done because the `cli.rs` file is not part of the crate: it is merely a binary. Thus, `cli.rs` can use code from our lib (like the testing code) but the reverse can't be done. Regardless, it makes more sense to me to have testing code in the test module, and, because we want to, to provide the same testing code via command-line, resembling the original `pconvert` |
| Screenshot | image |

### CODE REVIEW

**Review by gcandal — APPROVED** [review]

**Comment by joao-conde** [comment]
Re-assigning to me and all of the other `pconvert_rust` unit tests PR as CI started failing.

**Comment by joao-conde** [comment]
It is the same problem across all PRs: it is a compilation failure of `wasm-pack` (not a project dependency, just the tool used to build the wasm module) with rust version 1.40

**Comment by joamag** [comment]
@joao-conde please re-add the CLI features we want to make this compatible with the original convert, you can keep the test code. We can be redundant in the code no problem

**Comment by joamag** [comment]
adding fast-track to try to close this as soon as possible.
Let's also avoid unneeded refactor of the code, and instead use separate PRs for that

**Comment by joao-conde** [comment]
@joamag  the CLI test options still exist, just like in pconvert

File: src/cli.rs [hunk 1]
```diff
@@ -1,16 +1,12 @@
...
 use pconvert_rust::benchmark::Benchmark;
-use pconvert_rust::blending::{
-    blend_images, get_blending_algorithm, is_algorithm_multiplied, multiply_image, BlendAlgorithm,
-};
+use pconvert_rust::blending::BlendAlgorithm;
 use pconvert_rust::constants;
 use pconvert_rust::errors::PConvertError;
-use pconvert_rust::parallelism::{ResultMessage, ThreadPool};
-use pconvert_rust::utils::{read_png_from_file, write_png_parallel, write_png_to_file};
+use pconvert_rust::test::{apply_blue_filter, compose, compose_parallel, Background};
```
Comment (joamag) [inline src/cli.rs:7]: let's move this stuff from the test namespace and to something better like `utils` or `compose`

**Review by joamag — CHANGES_REQUESTED** [review]

**Comment by joamag** [comment]
@joao-conde move that stuff into a better place

**Comment by joao-conde** [comment]
Btw CI failing because of https://github.com/rustwasm/wasm-pack/issues/969

**Review by joamag — APPROVED** [review]

**Pull request merged**